### ISSUE TITLE
Is this for WSL1 and/or WSL2: Windows Support -> Windows Subsystem for Linux (WSL) -> Optional - Only if volumes are required:

### ISSUE THREAD

**Issue by cer** [testcontainers/testcontainers-java#5256]
This page https://www.testcontainers.org/supported_docker_environment/windows/ describes how to configure WSL to support volume mapping.

> Inside the WSL shell, modify the /ect/wsl.conf file to mount the Windows drives on / instead of on /mnt/. (Reboot required after this step).

However, it's not clear whether this applies to WSL1 or/and WSL2.

It would be helpful if this was clarified. There's also a typo.

**Comment by kiview** [comment]
Hey @cer, thanks for the pointer. This section only applied to *WSL1*. WSL2 should work, as documented under _WSL2 backend_. Would you like to provide a PR to correct this?

**Comment by kiview** [comment]
Does it work correctly if you use docker-compose directly?

**Comment by cer** [comment]
> Does it work correctly if you use docker-compose directly?

Yes.

**Comment by kiview** [comment]
Does it work correctly if you configure `DockerComposeContainer` using `withLocal(true)`?

**Comment by cer** [comment]
Unsure. my Windows machine is running Docker Compose 2 so I've been using  withLocal(false)

**Comment by kiview** [comment]
It would help to triage if you can try to use it with `withLocal(true)`, you can switch to Docker Compose 1 in the Docker Desktop settings.

But likely, this exact combination (Testcontainers with `DockerComposeContainer` file mounting and containerized Docker Compose on Docker Desktop with WSL2 backend from Windows process) does not work.

**Comment by cer** [comment]
Sorry for the delay. It works with `withLocalCompose(true)`.

I'm guessing the problem is with `convertToUnixFilesystemPath()`, which is called here: https://github.com/testcontainers/testcontainers-java/blob/66e4be2817f82aa80490c21e8df9e8f8a87fa792/core/src/main/java/org/testcontainers/containers/DockerComposeContainer.java#L623.
I believe it converts `c:\foo\bar` to `/c/foo/bar`.

**Comment by kiview** [comment]
You are right, this will be the problem. Especially in `DockerComposeContainer`, it is very hard to take care of all possible environment permutations. Would you like to look into providing a PR?

**Comment by janrieke** [comment]
We also just stumbled upon this issue.
Has there been some progress on this? Can we do something to help? If there is a snapshot with a fix, we'll be glad to give it a try.
